components/org.wso2.carbon.identity.api.server.common/src/main/java/org/wso2/carbon/identity/api/server/common/error/bulk/BulkErrorResponse.java

```diff
@@ -0,0 +1,118 @@
+import static org.wso2.carbon.identity.api.server.common.Util.isCorrelationIDPresent;
+
+/**
+ * Common BulkErrorResponse Object for all the server bulk API related errors.
+ */
+public class BulkErrorResponse extends BulkErrorDTO {
+
+    private static final long serialVersionUID = 373229260532218871L;
+
+    /**
+     * BulkErrorResponse Builder.
+     */
+    public static class Builder {
+        private String code;
+        private String message;
+        private String description;
+        private List<? extends ErrorDTO> failedOperations;
+
+        public Builder() {
+            // The constructor is intentionally left empty as there are no initializations required.
+        }
+
+        public BulkErrorResponse.Builder withCode(String code) {
+
+            this.code = code;
+            return this;
+        }
+
+        public BulkErrorResponse.Builder withMessage(String message) {
+
+            this.message = message;
+            return this;
+        }
+
+        public BulkErrorResponse.Builder withDescription(String description) {
+
+            this.description = description;
+            return this;
+        }
+
+        public BulkErrorResponse.Builder withFailedOperations(List<? extends ErrorDTO> failedOperations) {
+
+            this.failedOperations = failedOperations;
+            return this;
+        }
+
+        public BulkErrorResponse build() {
+
+            BulkErrorResponse error = new BulkErrorResponse();
+            error.setCode(this.code);
+            error.setMessage(this.message);
+            error.setDescription(this.description);
+            error.setRef(getCorrelation());
+            error.setFailedOperations(this.failedOperations);
+            return error;
+        }
+
+        public BulkErrorResponse build(Log log, Exception e, String message) {
+
+            BulkErrorResponse error = build();
+            String errorMessageFormat = "errorCode: %s | message: %s";
+            String errorMsg = String.format(errorMessageFormat, error.getCode(), message);
+            if (!isCorrelationIDPresent()) {
+                errorMsg = String.format("correlationID: %s | %s", error.getRef(), errorMsg);
+            }
+            log.error(errorMsg, e);
+            return error;
+        }
+
+        /**
+         * Build error response builder for bad requests without exceptions.
+         *
+         * @param log Logger.
+         * @param message Error message.
+         * @return BulkErrorResponse object.
+         */
+        public BulkErrorResponse build(Log log, String message) {
+
+            BulkErrorResponse error = build();
+            String errorMessageFormat = "errorCode: %s | message: %s";
+            String errorMsg = String.format(errorMessageFormat, error.getCode(), message);
+            if (!isCorrelationIDPresent()) {
+                errorMsg = String.format("correlationID: %s | %s", error.getRef(), errorMsg);
+            }
+
+            if (log.isDebugEnabled()) {
```

Comment:
The requirement to check if the debug logs enabled is to mainly avoid unnecessary string operations. So you have to move the `errorMsg` creation logic inside the if block.

Reply:
Moved the creation of `errorMsg` and `errorMessageFormat` inside the if block to avoid unnecessary object creation and string formatting when debug logs are disabled.